analysis/estimate/scripts/sync-data-files.R

```diff
@@ -0,0 +1,203 @@
+# This script prepares files needed for analysis:
+# 1. First, we identify samples of interest as non-multiplexed samples from solid tumors with paired bulk data
+# 2. Second, we sync all associated bulk `quant.sf` files and single-cell `_processed.rds` files from S3 needed for analysis
+# 3. Third, we sync all associated bulk counts from `_bulk_quant.tsv` from S3
+# All files are organized in `<project id>/<sample id>/` (except bulk counts which are only per project)
+# We also export two TSV file mapping files:
+# 1. A TSV mapping bulk library and sample ids, since the raw bulk counts use library id identifiers
+# 2. A TSV mapping gene symbols to ensembl ids, since ESTIMATE uses gene symbols
+
+
+renv::load()
+library(optparse)
+
+
+# Parse options --------
+option_list <- list(
+  make_option(
+    "--output_dir",
+    type = "character",
+    help = "Output directory to save synced data files organized by project/sample"
+  ),
+  make_option(
+    "--scpca_sce_file",
+    type = "character",
+    default = here::here("s3_files", "SCPCS000001", "SCPCL000001_processed.rds"),
+    help = "Path to an SCE file used to obtain mappings for converting ensembl ids to gene symbols"
+  ),
+  make_option(
+    "--library_sample_map_file",
+    type = "character",
+    help = "Path to output TSV file mapping bulk sample and library ids",
+  ),
```

Comment:
I know you do this in the other script, but I think you could skip this step here. We already have a script that you can run that will save the metadata for ScPCA to the `s3_files` folder (see https://github.com/AlexsLemonade/scpca-paper-figures/blob/main/scripts/figure_setup/sync-metadata.R). I'm not sure why you can't use the library metadata and just filter to only bulk IDs in later steps.